### PR TITLE
Added new and old CS majors

### PR DESCRIPTION
**Pull request by aryamanshodhan**
Implementation: BS CS major for old and new requirements

### CODE REVIEW

File: lib/resources/majors.tsx [hunk 1]
```diff
@@ -2255,17 +2255,12 @@ const bsCS_Old: Major = {
...
+            'At least 13 more upper level CS credits are required. ',
           required_credits: 13,
           criteria: 'EN Computer Science[D]^AND^Upper Level Undergraduate[L]',
           double_count: [
             '<b>Team Requirement</b> <br /> ' +
               'Select one course with Program of Study Tag CSCI-TEAM.',
-            '<b>CS Electives</b> <br /> ' +
-              'Eight additional credits of Computer Science are required.' +
-              'For an approved list of courses from other departments (maximum of 6 credits allowed), visit https://www.cs.jhu.edu/computer-science-other-courses-for-bs-degree/ ' +
-              'and create a custom course to satisfy this requirement.',
```
Comment (mpark63) [inline lib/resources/majors.tsx:2268]: `CS Electives` can overlap with `Upper Level CS Credits` but not vice versa?

File: lib/resources/majors.tsx [hunk 2]
```diff
@@ -2304,9 +2296,44 @@ const bsCS_Old: Major = {
...
+          required_credits: 3,
+          criteria: 'CSCI-RSNG[T]',
+          double_count: ['N/A'],
+        },
+      ],
+    },
+    {
+      name: 'Mathematics',
+      required_credits: 12,
+      min_credits_per_course: 4,
```
Comment (mpark63) [inline lib/resources/majors.tsx:2336]: criteria could be more specific? Currently probstat satisfies `Mathematics` when it should not 
AS.110.108
AS.110.109
EN.553.171

File: lib/resources/majors.tsx [hunk 3]
```diff
@@ -2304,9 +2296,44 @@ const bsCS_Old: Major = {
...
       description:
         'All courses in this category must be from one of the two math departments on ' +
         'campus: Mathematics or Applied Math and Statistics. However, 553.171 Discrete Mathematics ' +
```
Comment (mpark63) [inline lib/resources/majors.tsx:2339]: This also describes the `Mathematics Electives` distribution below.

File: lib/resources/majors.tsx [hunk 4]
```diff
@@ -2425,144 +2479,238 @@ const bsCS_New: Major = {
...
         {
           description:
-            '<b>Computer Ethics(601.104).</b><p>Practical Ethics for Future Leaders (660.400/406) may be used as a substitute for the computer ethics requirement for the BS program, but does not count towards the CS total credits at all.</p>',
+            '<b>Computer Ethics</b> <br /> Select one of the following courses: <br /> ' +
+            'EN.601.104 Computer Ethics <br /> ' +
+            'EN.601.124 The Ethics of Artificial Intelligence and Automation (The Ethics of Artificial Intelligence and Automation)' +
+            'EN.660.400 Practical Ethics for Future Leaders',
           required_credits: 1,
-          criteria: 'EN.600.104[C]^OR^EN.601.104[C]^OR^EN.660.400[C]',
+          criteria: 'EN.601.104[C]^EN.601.124[C]^OR^EN.660.400[C]',
```
Comment (mpark63) [inline lib/resources/majors.tsx:2498]: missing an ^OR^? 
`'EN.601.104[C]^OR^EN.601.124[C]^OR^EN.660.400[C]'`

File: lib/resources/majors.tsx [hunk 5]
```diff
@@ -2425,144 +2479,238 @@ const bsCS_New: Major = {
...
-          criteria:
-            'EN Computer Science[D]^AND^Upper Level Undergraduate[L]^NOT^EN.601.433[C]^NOT^EN.601.633[C]',
+          criteria: 'EN Computer Science[D]^AND^Upper Level[L]',
           double_count: [
-            '<b>One Team(CSCI-TEAM) designated course.</b><p> This Team course may overlap other course ' +
-              'requirements, for example to count as both Team and Software.</p>',
-            '<b>2 Upper Level Classifications:</b><p>At least one upper level course in two of these four different classification</p> ' +
-              'areas: Applications(CSCI-APPL), Systems(CSCI-SYST), Software(CSCI-SOFT) and Reasoning(CSCI-RSNG)',
+            '<b>Team Requirement</b> <br /> ' +
+              'Select one course with Program of Study Tag CSCI-TEAM.',
```
Comment (mpark63) [inline lib/resources/majors.tsx:2551]: can `Upper Level Undergraduate` not overlap with `CS Electives`?

File: lib/resources/majors.tsx [hunk 6]
```diff
@@ -2425,144 +2479,238 @@ const bsCS_New: Major = {
...
     {
-      name: 'Math',
-      required_credits: 16,
+      name: 'Computer Science Classifications',
+      required_credits: 6,
       min_credits_per_course: 3,
+      description:
+        'At least one course in two different classification areas (Applications, Reasoning, Software, Systems) must be chosen in addition to Theory (Algorithms).',
+      criteria: 'CSCI-APPL[T]^OR^CSCI-SOFT[T]^OR^CSCI-SYST[T]^OR^CSCI-RSNG[T]',
+      pathing: 2,
```
Comment (mpark63) [inline lib/resources/majors.tsx:2593]: double_count: [
        'Mathematics',
        'Humanities/Social Sciences',
        'Writing Intensive',
]

It's not a problem in frontend, but in backend there's no guarantee that distributions are processed in this order. So it is safer if all double_count fields are bidirectional. if A can double count with B, indicate that B can also double count with A. 

optional at this point but would appreciate it :)

File: lib/resources/majors.tsx [hunk 7]
```diff
@@ -2425,144 +2479,238 @@ const bsCS_New: Major = {
...
+    {
+      name: 'Mathematics',
+      required_credits: 8,
+      min_credits_per_course: 4,
       description:
         'All courses in this category must be from one of the two math departments on ' +
         'campus: Mathematics or Applied Math and Statistics. However, 553.171 Discrete Mathematics ' +
         'may not count towards these math requirements. Other than Calculus I and II, all the ' +
         'remaining courses must be 200-level or above.',
       criteria: 'AS Mathematics[D]^OR^EN Applied Mathematics & Statistics[D]',
```
Comment (mpark63) [inline lib/resources/majors.tsx:2630]: Criteria could be more specific so that prob stat doesn't end up here ^^

File: lib/resources/majors.tsx [hunk 8]
```diff
@@ -2425,144 +2479,238 @@ const bsCS_New: Major = {
...
+            'EN.553.311 Probability and Statistics for the Biological Sciences and Engineering',
           required_credits: 4,
           criteria: 'EN.553.211[C]^OR^EN.553.310[C]^OR^EN.553.311[C]',
         },
         {
           description:
-            '<p>2. Both Intro to Probability and Intro to Statistics</p><p>EN.553.420</p><p>EN.553.430</p> ',
+            '<b>Option 2: Probability and Statistics Separate </b> <br />' +
+            'EN.553.420 Introduction to Probability<br />' +
+            'EN.553.430 Introduction to Statistics<br />',
```
Comment (mpark63) [inline lib/resources/majors.tsx:2681]: awesome !

File: lib/resources/majors.tsx [hunk 9]
```diff
@@ -2425,144 +2479,238 @@ const bsCS_New: Major = {
...
       required_credits: 8,
       min_credits_per_course: 1,
       description:
-        'Students must take two semesters of core science courses (any combination of Physics, ' +
-        'Chemistry, Biology), with their associated labs. AP credit is an acceptable substitute for these courses and labs.',
+        'At least two semesters of physics or two semesters of chemistry, with the associated laboratories, must be included.',
       criteria: 'N[A]',
+      double_count: ['Humanities/Social Sciences', 'Writing Intensive'],
+      exception:
+        'EN Computer Science[D]^OR^AS Center for Language Education[D]^OR^EN Applied Mathematics & Statistics[D]',
```
Comment (mpark63) [inline lib/resources/majors.tsx:2696]: exception does not work. I think Olivia wanted to remove them?

File: lib/resources/majors.tsx [hunk 10]
```diff
@@ -2572,18 +2720,30 @@ const bsCS_New: Major = {
         'Students are required to fulfill the university’s requirement of two writing intensive courses, ' +
         'each at least 3 credits. Students must receive at least a C- grade or better in these writing courses. ',
       criteria: 'Written Intensive[W]',
```
Comment (mpark63) [inline lib/resources/majors.tsx:2722]: writing intensives can overlap with any distribution I think, but this works

File: lib/resources/majors.tsx [hunk 11]
```diff
@@ -2425,144 +2479,238 @@ const bsCS_New: Major = {
...
+        'Writing Intensive',
+      ],
       fine_requirements: [
         {
           description:
-            '<b>Computer Ethics(601.104).</b><p>Practical Ethics for Future Leaders (660.400/406) may be used as a substitute for the computer ethics requirement for the BS program, but does not count towards the CS total credits at all.</p>',
+            '<b>Computer Ethics</b> <br /> Select one of the following courses: <br /> ' +
+            'EN.601.104 Computer Ethics <br /> ' +
+            'EN.601.124 The Ethics of Artificial Intelligence and Automation (The Ethics of Artificial Intelligence and Automation)' +
+            'EN.660.400 Practical Ethics for Future Leaders',
```
Comment (mpark63) [inline lib/resources/majors.tsx:2496]: missing <br />

File: lib/resources/majors.tsx [hunk 12]
```diff
@@ -2425,144 +2479,238 @@ const bsCS_New: Major = {
...
+            '<b>Team Requirement</b> <br /> ' +
+              'Select one course with Program of Study Tag CSCI-TEAM.',
           ],
         },
         {
           description:
-            '<b>2 Upper Level Classifications:</b><p>At least one upper level course in two of these four different classification</p> ' +
-            'areas: Applications(CSCI-APPL), Systems(CSCI-SYST), Software(CSCI-SOFT) and Reasoning(CSCI-RSNG)',
+            '<b>CS Electives</b> <br /> ' +
+            'Six additional credits of Computer Science are required.' +
```
Comment (mpark63) [inline lib/resources/majors.tsx:2557]: super minor but insert space at the end haha

File: lib/resources/majors.tsx [hunk 13]
```diff
@@ -2425,144 +2479,238 @@ const bsCS_New: Major = {
...
+        "For more information please visit the <a href='https://www.cs.jhu.edu/undergraduate-studies/academics/ugrad-advising-manual/'>" +
         'major degree requirement</a> section on the department website.',
-      criteria:
-        'EN Computer Science[D]^OR^CSCI-OTHER[T]^OR^Gateway Computing[N]',
+      criteria: 'EN Computer Science[D]^OR^EN.500.112[C]^OR^EN.660.400[C]',
+      double_count: [
+        'Mathematics',
+        'Humanities/Social Sciences',
+        'Writing Intensive',
+      ],
```
Comment (mpark63) [inline lib/resources/majors.tsx:2489]: shouldn't it double count with Computer Science Classification also?

**Review by mpark63 — CHANGES_REQUESTED** [review]
Good work Aryaman! This is a major improvement from what we had before. The only changes I request are:
1. narrow down the criteria of `Mathematics` because `probstat` should not satisfy it, only Calc I and II should. 
2. I also think `Computer Science Classification` should overlap with `Computer Science` but I may be wrong. 

other comments are just typos. let me know if you have any questions.
again, great job!!! ⭐ ⭐⭐⭐⭐

**Review by mattliu-mygit — CHANGES_REQUESTED** [review]
From Elva:

Hey Matt, I think there’s a small bug with the CS degree check. In this plan(https://ucredit.me/share?_id=62d79a2eb1832a0004a7047d, I noticed that the degree checker isn’t including “Undergraduate Research EN.601.507” to the “CS Upper” requirement. It is also not including the “special cross-listed courses” as listed here(https://www.cs.jhu.edu/computer-science-other-courses-for-bs-degree/ by the CS department.

I just want to check that these extra CS other courses are accounted for and that we also have research and independent study satisfying this (they have course numbers as well).

File: lib/resources/majors.tsx (hunk 1 above)
Comment (aryamanshodhan) [reply lib/resources/majors.tsx:2268]: Should not be possible either way. Electives need to be exclusive from everything else

File: lib/resources/majors.tsx (hunk 2 above)
Comment (aryamanshodhan) [reply lib/resources/majors.tsx:2336]: Addressed

File: lib/resources/majors.tsx (hunk 3 above)
Comment (aryamanshodhan) [reply lib/resources/majors.tsx:2339]: Done

File: lib/resources/majors.tsx (hunk 4 above)
Comment (aryamanshodhan) [reply lib/resources/majors.tsx:2498]: Addressed

File: lib/resources/majors.tsx (hunk 5 above)
Comment (aryamanshodhan) [reply lib/resources/majors.tsx:2551]: No

File: lib/resources/majors.tsx (hunk 6 above)
Comment (aryamanshodhan) [reply lib/resources/majors.tsx:2593]: Added and addressed

File: lib/resources/majors.tsx (hunk 8 above)
Comment (aryamanshodhan) [reply lib/resources/majors.tsx:2681]: ty!!

File: lib/resources/majors.tsx (hunk 7 above)
Comment (aryamanshodhan) [reply lib/resources/majors.tsx:2630]: Addressed

File: lib/resources/majors.tsx (hunk 9 above)
Comment (aryamanshodhan) [reply lib/resources/majors.tsx:2696]: Done

File: lib/resources/majors.tsx (hunk 10 above)
Comment (aryamanshodhan) [reply lib/resources/majors.tsx:2722]: I thought not including anything means by default everything works

File: lib/resources/majors.tsx (hunk 11 above)
Comment (aryamanshodhan) [reply lib/resources/majors.tsx:2496]: What is missing?

File: lib/resources/majors.tsx (hunk 12 above)
Comment (aryamanshodhan) [reply lib/resources/majors.tsx:2557]: Done

File: lib/resources/majors.tsx (hunk 13 above)
Comment (aryamanshodhan) [reply lib/resources/majors.tsx:2489]: You're right. Done

**Review by mattliu-mygit — APPROVED** [review]

File: lib/resources/majors.tsx (hunk 11 above)
Comment (mpark63) [reply lib/resources/majors.tsx]: `<br />`

File: lib/resources/majors.tsx (hunk 1 above)
Comment (mpark63) [reply lib/resources/majors.tsx:2268]: ohh my bad I was looking at the wrong changes

**Review by mpark63 — APPROVED** [review]

**Pull request merged**